### ISSUE TITLE
Mono approximation bug?

### ISSUE THREAD

**Issue by pgkirsch** [hoburg/gpkit#534]
```python

In [1]: from gpkit import Variable

In [2]: x = Variable('x')

In [3]: p = 1 + x

In [4]: p.mono_approximation({x:1})
Out[4]: gpkit.Monomial(2*x**0.5)

In [5]: p = 1 + 1/x

In [6]: p.mono_approximation({x:1})
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-6-22bfe42df921> in <module>()
----> 1 p.mono_approximation({x:1})

/Users/philippekirschen/Documents/MIT/Research/GPkit/gpkit/gpkit/nomials.pyc in mono_approximation(self, x_0)
    179         m0 = 1
    180         for vk in self.varlocs:
--> 181             e = mag(x_0[vk]*self.diff(vk).sub(x_0, require_positive=False).c/p0)
    182             exp[vk] = e
    183             m0 *= (x_0[vk])**e

/Users/philippekirschen/Documents/MIT/Research/GPkit/gpkit/gpkit/nomials.pyc in diff(self, wrt)
    152         """
    153         deriv = super(Signomial, self).diff(wrt)
--> 154         return Signomial(exps=deriv.exps, cs=deriv.cs)
    155 
    156     def mono_approximation(self, x_0):

/Users/philippekirschen/Documents/MIT/Research/GPkit/gpkit/gpkit/nomials.pyc in __init__(self, exps, cs, require_positive, simplify, **descr)
    109             from . import SIGNOMIALS_ENABLED
    110             if require_positive and not SIGNOMIALS_ENABLED:
--> 111                 raise ValueError("each c must be positive.")
    112         else:
    113             self.__class__ = Posynomial

ValueError: each c must be positive.
```

**Comment by bqpd** [comment]
Well. it ain't the wrong approximation, it just isn't a proper monomial..

**Comment by bqpd** [comment]
The correct approximation is a line with negative slope.

**Comment by bqpd** [comment]
Wait hmm.

**Comment by whoburg** [comment]
Agreed this is a bug.

It's because `diff` needs to compute a signomial with negative cs as an intermediate step to computing this mono_approximation.

The user should not have to use `SignomialsEnabled` here. We could `require_positive=False` for all calls to diff? That seems a bit scary though

**Comment by bqpd** [comment]
Why? It's not really supposed to be a signomial... (and it does so in mono_approx)

**Comment by whoburg** [comment]
@pgkirsch, this was fixed by #545.

New output below:
```python
In [1]: from gpkit import Variable

In [2]: x = Variable('x')

In [3]: p = 1 + x

In [4]: p.mono_approximation({x: 1})
Out[4]: gpkit.Monomial(2*x**0.5)

In [5]: p = 1 + 1/x

In [6]: p.mono_approximation({x: 1})
Out[6]: gpkit.Monomial(2*x**-0.5)
```
Please close this issue if satisfied.

**Comment by pgkirsch** [comment]
Looks good, thanks!
